src/fpm_environment.f90

```diff
@@ -154,26 +154,67 @@ logical function os_is_unix(os) result(unix)
         else
             build_os = get_os_type()
         end if
-        unix = os /= OS_WINDOWS
+        unix = build_os /= OS_WINDOWS
     end function os_is_unix
 
     !> echo command string and pass it to the system for execution
-    subroutine run(cmd,echo,exitstat)
+    subroutine run(cmd,echo,exitstat,verbose,redirect)
         character(len=*), intent(in) :: cmd
         logical,intent(in),optional  :: echo
         integer, intent(out),optional  :: exitstat
-        logical :: echo_local
-        integer :: stat
+        logical, intent(in), optional :: verbose
+        character(*), intent(in), optional :: redirect
+
+        logical :: echo_local, verbose_local
+        character(:), allocatable :: redirect_str
+        character(1000) :: line
```

Comment:
This is dangerous, I think we already have a readline/getline procedure for reading into a deferred length character around.

Reply:
Oops, didn't spot that. What do you advise for how to use `fpm_filesystem::getline` in `fpm_environment` without a circular dependency?

Reply:
Can we move `getline` or this procedure to get away without a cyclic dependency?

Reply:
Ah yeah okay, I'll probably move `fpm_environment::run` to `fpm_filesystem` then.